backend/lucene/src/main/java/org/hibernate/search/backend/lucene/document/model/impl/LuceneIndexSchemaFieldNode.java

```diff
@@ -94,11 +96,20 @@ public String toString() {
 				.append( ", predicateBuilderFactory=" ).append( predicateBuilderFactory )
 				.append( ", sortContributor=" ).append( sortBuilderFactory )
 				.append( ", projectionBuilderFactory=" ).append( projectionBuilderFactory )
+				.append( ", indexNullAsValue=" ).append( indexNullAsValue )
 				.append( "]" );
 		return sb.toString();
 	}
 
 	private EventContext getEventContext() {
 		return EventContexts.fromIndexFieldAbsolutePath( absoluteFieldPath );
 	}
+
+	public void encode(AbstractLuceneDocumentBuilder documentBuilder, String absoluteFieldPath, F value) {
+		if ( value == null && indexNullAsValue.isPresent() ) {
+			value = indexNullAsValue.get();
+		}
```

Comment:
To me it would make more sense to implement this in the codec itself... There's already a `if ( value == null )` block there, that would be the ideal place.

If you really want to put this in the schema node, which was initially supposed to simply hold metadata, I won't force your hand, but I suspect we'll have to revert it at some point.